vuestorefront/models/product_template.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 ODOOGAP/PROMPTEQUATION LDA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, fields, api
+from odoo.tools.sql import column_exists, create_column
+
+from odoo.addons.http_routing.models.ir_http import slug, slugify
+
+
+def slugify_with_sfx(value, sfx):
+    return f"{slugify(value)}{sfx}"
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def _auto_init(self):
+        """Create/set website_slug via SQL to save on memory usage."""
+        cr = self.env.cr
+        if not column_exists(
+            cr, "product_template", "website_slug"
+        ):
+            create_column(
+                cr, "product_template", "website_slug", "varchar"
+            )
+            cr.execute("SELECT id, name FROM product_template WHERE name IS NOT NULL")
+            for res in cr.fetchall():
+                id_, name = res
+                cr.execute(
+                    "UPDATE product_template SET website_slug = %s WHERE id = %s",
+                    (slugify_with_sfx(name, f"-{id_}"), id_)
+                )
+        return super()._auto_init()
+
+    def _get_public_categ_slug(self, category_ids, category):
+        category_ids.append(category.id)
+
+        if category.parent_id:
+            category_ids = self._get_public_categ_slug(category_ids, category.parent_id)
+
+        return category_ids
+
+    @api.depends('public_categ_ids')
+    def _compute_public_categ_slug_ids(self):
+        """ To allow search of website_slug on parent categories """
+        cr = self.env.cr
+
+        for product in self:
+            category_ids = []
+
+            for category in product.public_categ_ids:
+                category_ids = product._get_public_categ_slug(category_ids, category)
+
+            cr.execute("""
+                DELETE FROM product_template_product_public_category_slug_rel
+                WHERE product_template_id=%s;
+            """, (product.id,))
+
+            for category_id in list(dict.fromkeys(category_ids)):
+                cr.execute("""
+                    INSERT INTO product_template_product_public_category_slug_rel(product_template_id, product_public_category_id)
+                    VALUES(%s, %s);
+                """, (product.id, category_id,))
+
+    website_slug = fields.Char(
+        'Website Slug',
+        compute='_compute_website_slug',
+        store=True,
+        readonly=True,
+        index=True,
+    )
+    public_categ_slug_ids = fields.Many2many('product.public.category',
```

Comment:
I thought we agreed that the name is misleading and need to rename it. Also, this will fail the installation of the module of a larger amount of products.